syntax = "proto3";

import "google/protobuf/struct.proto";
import "google/protobuf/timestamp.proto";


message Offer {
    string id = 1;
    Apartment apartment = 2;
    optional string title = 3;
    optional string description = 4;
    optional google.protobuf.Timestamp time_scraped = 5;
}

message Apartment {
    Money price = 1;
    DecimalNumber size = 2;
    optional LatLang location = 3;
    optional string address = 4;
    optional int32 year_built = 5;
    optional int32 room_count = 6;
    optional int32 floor = 7;
    map<string, google.protobuf.Value> properties = 8;
}

message LatLang {
    double latitude = 1;
    double longitude = 2;
}

message Money {
    DecimalNumber value = 1;
    Currency currency = 2;
}

message DecimalNumber {
    int64 value = 1;
    optional int32 scale = 2;  
}

enum Currency {
    CURRENCY_UNSPECIFIED = 0;
    PLN = 1;
}
